src/internal_rep.rs

```diff
@@ -1338,6 +1348,277 @@ fn call_to_fc_rules<'a>(
+                is_list_param: true,
+                name: CascadeString::from("file_type"),
+                default: Some(Argument::List(vec![])),
+            },
+            types,
+            None,
+        )?,
+    ];
+    let validated_args = validate_arguments(c, &target_args, types, class_perms, context, file)?;
+    let mut args_iter = validated_args.iter();
+    let mut ret = Vec::new();
+
+    let context_str = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?;
+    let fs_context = match Context::try_from(context_str.to_string()) {
+        Ok(c) => c,
+        Err(_) => {
+            return Err(CascadeErrors::from(ErrorItem::Compile(CompileError::new(
+                "Invalid context",
+                file,
+                context_str.get_range(),
+                "Cannot parse this into a context",
+            ))))
+        }
+    };
+    let fs_name = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?
+        .to_string();
+    let fscontext_str = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?;
+    let fscontext_type = match fscontext_str.to_string().parse::<FSContextType>() {
+        Ok(f) => f,
+        Err(_) => {
+            return Err(CascadeErrors::from(ErrorItem::Compile(CompileError::new(
+                "Not a valid file system type",
+                file,
+                fscontext_str.get_range(),
+                "File system type must be 'xattr', 'task', 'trans', or 'genfscon'",
+            ))));
+        }
+    };
+    let regex_string_arg = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?;
+    let regex_string = regex_string_arg.get_name_or_string(context)?.to_string();
+    let file_types_arg = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?;
+    let file_types = file_types_arg.get_list(context)?;
+
+    match fscontext_type {
+        FSContextType::XAttr | FSContextType::Task | FSContextType::Trans => {
+            // The 'regex_string_arg.get_range().is_none()' is a hacky way to
+            // to check if arg was actually provided or not.  Since we set the
+            // default for regex_string to "/" this is the only way I could find
+            // to test if the actual arg was passed or not
+            if regex_string_arg.get_range().is_none() && file_types.is_empty() {
+                ret.push(FileSystemContextRule {
+                    fscontext_type,
+                    fs_name,
+                    path: None,
+                    file_type: None,
+                    context: fs_context.clone(),
+                });
+            } else if !file_types.is_empty() {
```

Comment:
I think that if a user has made both mistakes we should return both errors at once.